src/Cardano/Shell/NodeIPC/Lib.hs

```diff
@@ -162,44 +176,58 @@ getIPCHandle = do
             Right fd -> liftIO $ fdToHandle fd
 
 -- | Start IPC with given 'ReadHandle', 'WriteHandle' and 'Port'
-startIPC :: forall m. (MonadIO m) => ReadHandle -> WriteHandle -> Port -> m ()
-startIPC readHandle writeHandle port = liftIO $ void $ ipcListener readHandle writeHandle port
+startIPC :: forall m. (MonadIO m) => ProtocolDuration -> ReadHandle -> WriteHandle -> Port -> m ()
+startIPC protocolDuration readHandle writeHandle port = liftIO $ void $ ipcListener protocolDuration readHandle writeHandle port
 
 -- | Start IPC with NodeJS
 --
 -- This only works if NodeJS spawns the Haskell executable as child process
 -- (See @server.js@ as an example)
-startNodeJsIPC :: forall m. (MonadIO m) => Port -> m ()
-startNodeJsIPC port = do
-    handle <- liftIO $ getIPCHandle
-    let readHandle = ReadHandle handle
+startNodeJsIPC :: forall m. (MonadIO m) => ProtocolDuration -> Port -> m ()
+startNodeJsIPC protocolDuration port = do
+    handle          <- liftIO $ getIPCHandle
+    let readHandle  = ReadHandle handle
     let writeHandle = WriteHandle handle
-    liftIO $ void $ ipcListener readHandle writeHandle port
+    liftIO $ void $ ipcListener protocolDuration readHandle writeHandle port
+
+-- | Function for handling the protocol
+handleIPCProtocol :: forall m. (MonadIO m) => Port -> MsgIn -> m MsgOut
```

Comment:
The protocol handler function, which can be extended.
The `Port` is there because it's the simplest way to include it.

Reply:
Does extended mean you can have the shell to perform any arbitrary `IO` action when handling the message? If so, this LGTM.
At glance I didn't understand why this function has MonadIO constraint

Reply:
> At glance I didn't understand why this function has MonadIO constraint

It's there to enable adding any kind of message to the protocol.
So if somebody first needs to fetch something or do something with `IO` before returning a response, they have nothing to fear (except the amount of freedom).